src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -3,27 +3,82 @@
  */
 #pragma once
 
+#include "ai/world/ball.h"
 #include "ai/world/robot.h"
+#include "geom/angle.h"
 #include "geom/point.h"
 #include "geom/polygon.h"
 #include "geom/util.h"
+#include "shared/constants.h"
 
 class Obstacle
 {
    public:
+    static Obstacle createPlaceholderObstacle();
+
+    static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
+
     /*
      * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer scaled by radiusScaling and
-     * velocity_projection_scaling
+     * should not enter with a buffer scaled by radius_scaling and
+     * velocity_cushion_scaling
      *
      * @param robot robot to create obstacle boundary polygon around
-     * @param radiusScaling safety factor to vary the radius size of the buffer zone
-     * (1=default), must be greater than 0
-     * @param velocity_projection_scaling scales the projection buffer in the direction of
-     * the velocity (1=default), must be greater than 0
+     * @param radius_scaling multiplicatively scales the radius of the obstacle
+     * @param velocity_cushion_scaling multiplicatively scales the cushion of
+     * velocity of the robot
+     * @param enable_velocity_cushion if true, enables the cushion of the velocity
+     * of the robot as component of the obstacle
```

Comment:
This boolean seems redundant, seeing as we can just set the `velocity_cushion_scaling` to 0 above (presuming I'm understanding this all correctly).

Reply:
I think that the velocity cushion scaling should always be greater than 1 in this case. This API is consistent with the buffer params API and I think it should be.

Reply:
Sorry, I still don't quite follow. Why can't we just set the `velocity_cushion_scaling` to be one? Also, what API are you referring to?

Reply:
ok fine I'll change it